run.py

```diff
@@ -357,30 +466,59 @@ def handle_test(self, test_index: int, test: Dict[str, Any], runner: str, versio
             if progress:
                 print(f"Running test {test_index} (ID: {test['id']}) with runner {runner} on WDL version {version}.")
             response.update(
-                self.run_single_test(test_index, test, runner, version, time, verbose, quiet, args, jobstore_path))
+                self.run_single_test(test_index, test, runner, version, time, verbose, quiet, args, jobstore_path, debug))
         if repeat is not None:
             response["repeat"] = repeat
+        # Turn failing tests to warnings if they violate a test dependency
+        response.update(test_dependencies(dependencies=test.get("dependencies"), current_result=response))
```

Comment:
If we're passing response in, and making this function return a modification, and then applying the modification, I think it gets kind of confusing.

We could make the function take the old response and return a new modified response copy. Or we could make it modify the response in place. Either of those would be simpler, and maybe easier to explain in the docstring.